business/checkers/destinationrules/traffic_policy_checker.go

```diff
@@ -0,0 +1,102 @@
+package destinationrules
+
+import (
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/models"
+)
+
+type TrafficPolicyChecker struct {
+	DestinationRules []kubernetes.IstioObject
+	MTLSDetails      kubernetes.MTLSDetails
+}
+
+func (t TrafficPolicyChecker) Check() models.IstioValidations {
+	validations := models.IstioValidations{}
+
+	// When mTLS is not enabled, there is no validation to be added.
+	if !t.hasEnablingmTLSDR() {
+		return validations
+	}
+
+	// Check whether DRs override mTLS.
+	for _, dr := range t.DestinationRules {
+		if !hasTrafficPolicy(dr) || !hasmTLSSettings(dr) {
+			check := models.Build("destinationrules.trafficpolicy.notlssettings", "spec/trafficPolicy")
+			key := models.BuildKey(DestinationRulesCheckerType, dr.GetObjectMeta().Name)
+			validation := buildDestinationRuleValidation(dr, check, true)
+
+			if _, exists := validations[key]; !exists {
+				validations.MergeValidations(models.IstioValidations{key: validation})
+			}
+		}
+	}
+
+	return validations
+}
+
+func (t TrafficPolicyChecker) hasEnablingmTLSDR() bool {
+	enablesTLS := false
+
+	for _, dr := range t.MTLSDetails.DestinationRules {
+		enablesTLS = enablesmTLS(dr)
```

Comment:
Wouldn't it be simpler to do:

```go
for _, dr := range t.MTLSDetails.DestinationRules {
if enablesmTLS(dr) { return true }
}
return false
```

Nitpicking .. ;)

Reply:
Always open to this kind of things. Specially if they follow go's idiosyncrasy. :+1: